src/environment.cpp

```diff
@@ -172,21 +172,22 @@ void Environment::continueRaycast(RaycastState *state, PointedThing *result)
 			bool is_colliding = false;
 			// Minimal distance of all collisions
 			float min_distance_sq = 10000000;
+			// ID of the current box (loop counter)
+			u16 id = 0;
 
 			v3f npf = intToFloat(np, BS);
-			for (std::vector<aabb3f>::const_iterator i = boxes.begin();
-					i != boxes.end(); ++i) {
-				// Get current collision box
-				aabb3f box = *i;
+			for (aabb3f &box : boxes) {
```

Comment:
wrong reference, previously we copied the object

Reply:
This isn't a bug, `boxes` is not used after the loop. I just prevented unnecessary copying.